framework/resource.go

```diff
@@ -0,0 +1,21 @@
+package framework
+
+import "context"
+
+// Resource are the building blocks of the operator's reconciliation logic.
+// Note there can be multiple Resources reconciling the same object in the
+// chain. In that case they are guaranteed to be executed in order one one
+// after another.
+type Resource interface {
+	// Name returns the resource's name used for identification.
+	Name() string
+
+	// EnsureCreated is called whtn observed object is created or updated.
+	// The object is in state after cration or modification. This method must
+	// be idempotent.
+	EnsureCreated(ctx context.Context, obj interface{}) error
+	// EnsureDeleted is called whtn observed object is deleted. The object
+	// is in last observed state before the deletion. This method must be
+	// idempotent.
+	EnsureDeleted(ctx context.Context, obj interface{}) error
+}
```

Comment:
Brand new simplified Resource interface.

Reply:
Shouldn't be interfaces defined in spec files?

Reply:
I can put it there, but it isn't Framework interface, so it felt wrong. WDYT?

Reply:
In general looks like the convention is putting them in spec files, is that correct? Previously it was that way. I've also seen cases in which more than one interface was defined in a spec file, not only one per package. Anyway maybe in this case doesn't make too much sense, I trust your feelings, as you see fit :)

Reply:
Then should we also put `CRUDResourceOps` in `spec.go` but `CRUDResource` in `crud_resource.go` because it isn't interface? If we had a main struct for the package like `Framework` in this case it would make sense to put `type Interface interface` in spec.go I agree.